docs/src/man/basics.md

```diff
@@ -1510,32 +1510,95 @@ julia> german[Not(5), r"S"]
-  In this scenario we apply the transformation function to `source_column` and the
-  target column names is automatically generated.
-- `source_column => target_column_name` renames the `source_column` to `target_column_name`.
-- `source_column` just keep the source column as is in the result without any transformation;
-
-These rules are typically called transformation mini-language.
-
+In DataFrames.jl there are seven functions that can be used to transform data frame columns:
+
+| Function     | Memory Usage                     | Column Retention | Row Retention |
+| ------------ | -------------------------------- | ---------------- | ------------ |
+| `transform`  | Creates a new data frame.        | Retains both source and transformed columns. | Retains same number of rows as source data frame. |
+| `transform!` | Modifies an existing data frame. | Retains both source and transformed columns. | Retains same number of rows as source data frame. |
+| `select`     | Creates a new data frame.        | Retains only transformed columns. | Retains same number of rows as source data frame. |
+| `select!`    | Modifies an existing data frame. | Retains only transformed columns. | Retains same number of rows as source data frame. |
+| `subset`     | Creates a new data frame.        | Retains both source and transformed columns. | Number of rows may change.                        |
+| `subset!`    | Modifies an existing data frame. | Retains both source and transformed columns. | Number of rows may change.                        |
+| `combine`    | Creates a new data frame.        | Retains only transformed columns. | Number of rows may change.                        |
+
+### Constructing Transformation Pairs
+All of the functions above use the same syntax which is commonly
+`transformation_function(source_dataframe, transformation)`.
+The `transformation` argument is a `Pair` which defines the
+transformation to be applied to the `source_dataframe`,
+and it can take any of the forms listed below:
+
+- `source_column_selector => function`
+  - passes source column(s) to function
+  - automatically names the resulting column(s)
+- `source_column_selector => function => new_column_names`
+  - passes source column(s) to function
+  - names the resulting column(s) `new_column_names`
+  - *cannot be used with `subset` or `subset!`*
+- `source_column_selector => new_column_names`
+  - renames source column
+  - *may only select one column unless broadcasting*
+  - *cannot be used with `subset` or `subset!`*
+- `source_column_selector`
+  - selects source column(s) without transforming them
+  - *often used with `select` or `select!` for isolating or moving columns*
+
+#### `source_column_selector`
+The most basic `source_column_selector` is a column name, but there are many more ways to
+select columns as explained in the first section of the 
+[Indexing API documentation](https://dataframes.juliadata.org/stable/lib/indexing/#Indexing).
```

Comment:
How do I make this an internal link?

Reply:
Please check the Documenter.jl documentation. I never remember this. Maybe just writing `#Indexing` is enough?

Reply:
I think `[Indexing](@ref)`? https://juliadocs.github.io/Documenter.jl/dev/man/syntax/#@ref-link